clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -987,4 +1052,26 @@ public int compare(ConfigKey k1, ConfigKey k2) {
         });
         return configs;
     }
+
+    /**
+     * Get a list of configs sorted into "group" order: listing group, then
+     * ordering by order in group.
+     */
+    private List<ConfigKey> sortedConfigsInGroup() {
+        // sort first with group, then by name
+        List<ConfigKey> configs = new ArrayList<>(this.configKeys.values());
+        Collections.sort(configs, new Comparator<ConfigKey>() {
+            public int compare(ConfigKey k1, ConfigKey k2) {
+                // sort by group
+                int cmp = k1.group.compareTo(k2.group);
+                if (cmp == 0) {
+                    // then sort with order in group
+                    return k1.orderInGroup >= k2.orderInGroup ? -1 : 1;
```

Comment:
Currently this is reversing the order. Probably easier to just do `return Integer.compare(k1.orderInGroup, k2.orderInGroup)` here.